### ISSUE TITLE
Uncaught TypeError: Cannot read property 'previousSibling' of null

### ISSUE THREAD

**Issue by dadisn** [derbyjs/saddle#20]
Saddle throws an error when trying to append an each block containing nothing. This situation can occur for example when filtering out items with an if sentence from a list being displayed with an each block.

```html
{{each list as #item}}
  {{if #item.show}}
    <p>{{#item}}</p>
  {{/if}}
{{/each}}
```

The error occurs when trying to access `node.previousSibling` on a `node` that is `null`. See https://github.com/derbyjs/saddle/blob/master/index.js#L819.

See https://github.com/dadisn/derby-bugreport for how to recreate the bug. 

Screenshot of bug and error message can be found here: https://www.dropbox.com/s/s9e6mmz4tbb3avj/Screenshot%202015-01-12%2013.33.45.png?dl=0

**Comment by ovvn** [comment]
The bug occurs to me when filtering inside an each loop. 
Example
```jade
each #games as #game
 if #game.mode === "unpublished"
   p test
```

**Comment by dadisn** [comment]
Correct, this occurs when there is no content in the `each` loop to be rendered - either because it is plainly empty or because all the content is being filtered away.

**Comment by nateps** [comment]
Thanks for figuring this out and the repro case!

**Comment by nateps** [comment]
Fixed by 5eac3fd. The code actually dealt with the null case correctly except that I didn't properly check in accessing the property. Published as Saddle 0.5.6

**Comment by dadisn** [comment]
Thanks for a quick response and fix!
